core/enumerator.rbs

```diff
@@ -124,6 +124,43 @@
 class Enumerator[unchecked out Elem, out Return] < Object
   include Enumerable[Elem]
 
+  # <!--
+  #   rdoc-file=enumerator.c
+  #   - Enumerator.produce(initial = nil) { |prev| block } -> enumerator
+  # -->
+  #
+  # Creates an infinite enumerator from any block, just called over and
+  # over.  The result of the previous iteration is passed to the next one.
+  # If +initial+ is provided, it is passed to the first iteration, and
+  # becomes the first element of the enumerator; if it is not provided,
+  # the first iteration receives +nil+, and its result becomes the first
+  # element of the iterator.
+  #
+  # Raising StopIteration from the block stops an iteration.
+  #
+  #   Enumerator.produce(1, &:succ)   # => enumerator of 1, 2, 3, 4, ....
+  #
+  #   Enumerator.produce { rand(10) } # => infinite random number sequence
+  #
+  #   ancestors = Enumerator.produce(node) { |prev| node = prev.parent or raise StopIteration }
+  #   enclosing_section = ancestors.find { |n| n.type == :section }
+  #
+  # Using ::produce together with Enumerable methods like Enumerable#detect,
+  # Enumerable#slice_after, Enumerable#take_while can provide Enumerator-based alternatives
+  # for +while+ and +until+ cycles:
+  #
+  #   # Find next Tuesday
+  #   require "date"
+  #   Enumerator.produce(Date.today, &:succ).detect(&:tuesday?)
+  #
+  #   # Simple lexer:
+  #   require "strscan"
+  #   scanner = StringScanner.new("7+38/6")
+  #   PATTERN = %r{\d+|[-/+*]}
+  #   Enumerator.produce { scanner.scan(PATTERN) }.slice_after { scanner.eos? }.first
+  #   # => ["7", "+", "38", "/", "6"]
+  def self.produce: (?untyped initial) { (untyped prev) -> untyped } -> self
```

Comment:
At first, I thought it would be better to use a type parameter for this method (ex. `[T] (T initial) { (T prev) -> T } -> self[T, untyped]`).
But developers can use the different types between `initial` and `prev` (the return type of the block). So it's difficult to give a better type to this method to me. Therefore, I used `untyped` for this.